### DISCUSSION TITLE
Routeless Routing:  Manual user path matching

### DISCUSSION THREAD

**Discussion by yaq-cc** [encode/starlette#2098] [Q&A] [answered]
Is it possible to process requests directly using Starlette without using a routing table and if so, how?

For instance (in Go), it's possible to handle a request directly with a Handler and then perform path based routing manually:

```go
func ManualRouteHandler(w http.ResponseWriter, req *http.Request) {
	cleanPath := path.Clean(req.URL.Path)
	if strings.HasPrefix(cleanPath, "/api") {
		// if the the path starts with /api, use the apiHandler
		// pass down the response and request to the next handler
		apiHandler.ServeHTTP(w, req)
		return
	} else {
		// if it's any other path, use the fileHandler
		// pass down the response and request to the next handler
		fileHandler.ServeHTTP(w, req)
		return
	}
}
```

Thank you!

**Comment by adriangb** [comment, marked as answer]
Yes and no: you can easily create a catchall handler (`"/{rest_of_path:path}"` or `/foo/{rest_of_path:path}`) and handle routing in there. But there is a still a routing table running. If you want no routing table whatsoever you can crate a pure ASIG app and use Starlette as a toolkit:

```python
async def app(scope, receive, send):
    request = Request(scope, receive)
    ...
```
